beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs

```diff
@@ -752,23 +779,34 @@ impl<T: BeaconChainTypes> OverflowLRUCache<T> {
         }
     }
 
+    fn block_import_requirement(
+        &self,
+        pending_components: &PendingComponents<T::EthSpec>,
+    ) -> Result<BlockImportRequirement, AvailabilityCheckError> {
+        let epoch = pending_components
+            .epoch()
+            .ok_or(AvailabilityCheckError::UnableToDetermineImportRequirement)?;
```

Comment:
Is this error necessary? The import requirements are not complete until the block shows up, so you can assume that it will always eventually show up. Instead of returning an error here, you can return something variant signaling that requirements are not known yet.

Reply:
Good question - I thought about this and my reason was that `pending_components.epoch` should always return an epoch unless there's a bug - there should be at least one component in `pending_components` when we're checking availability, and any component (block/blob/column) would have a slot / epoch attached to it.

I think it's safer to return an error here, in case if there's a bug in `pending_components.epoch()` - if we've already received all necessary components, this would never be called again and we end up having stuck pending components.